### PR TITLE
Login frontend 2

### PR DESCRIPTION
**Pull request by ErikEmmanuelAnrique**
### **Summary**

Implemented basic validation for user data. Main Activity changed from _LoginActivity_ to _RootActivity_.

### **Screenshots**

Screenshot_1616433850

### **Trello Card**

https://trello.com/c/ZAAmG3I4/17-u1-frontend-login-validaciones

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/login/LoginPresenter.kt [hunk 1]
```diff
@@ -1,6 +1,37 @@
...
 import javax.inject.Inject
 
-class LoginPresenter @Inject constructor() : BasePresenter<LoginView>()
\ No newline at end of file
+class LoginPresenter @Inject constructor(private val userSession: UserSession) : BasePresenter<LoginView>() {
+
+    private fun isEmailValid(email: String): Boolean {
+
+        return android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()
+    }
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/login/LoginPresenter.kt]: Move this private function to the bottom of the file

File: app/src/main/java/ar/com/wolox/android/login/LoginPresenter.kt [hunk 2]
```diff
@@ -1,6 +1,37 @@
...
+            userSession.email = email
+            userSession.password = password
+        } else {
+
+            for (actualError in totalErrors) {
+
+                actualError.callAction(view!!)
+            }
+        }
+    }
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/login/LoginPresenter.kt]: We usually prefer to have curly brackets in the if/else statements, even if they are a single line. Like:
```
if (email.isEmpty()) {
    totalErrors.add(Errors.EMPTY_EMAIL)
}
```

File: app/src/main/java/ar/com/wolox/android/login/utils/Errors.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    EMPTY_PASSWORD {
+        override fun callAction(view: LoginView) { view?.showEmptyPasswordError() }
+    },
+    EMPTY_EMAIL {
+        override fun callAction(view: LoginView) { view?.showEmptyEmailError() }
+    },
+    INVALID_EMAIL {
+        override fun callAction(view: LoginView) { view?.showInvalidEmailError() }
+    };
+    abstract fun callAction(view: LoginView)
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/login/utils/Errors.kt]: I would make `view` nullable. Like `view: LoginView?`

**Review by ffasitella — CHANGES_REQUESTED** [review]
Wow! Amazing job!!
Just some little changes and we are ready to go! 🚀

**Review by jcruz-wolox — APPROVED** [review]
If the Fede's changes are implemented, this PR is approved for me.
Good job!

**Review by ffasitella — APPROVED** [review]
Good job! 👍🏽

**Pull request merged**